crates/storage/provider/src/providers/database/mod.rs

```diff
@@ -43,7 +41,7 @@ pub struct ProviderFactory<DB> {
     /// Chain spec
     chain_spec: Arc<ChainSpec>,
     /// Snapshot Provider
-    snapshot_provider: Option<Arc<SnapshotProvider>>,
+    pub snapshot_provider: Option<Arc<SnapshotProvider>>,
```

Comment:
please keep this private, mixing pub and private fields is not great

Reply:
agreed, there's already a method for that